File: server/internal/dto/theme.go
package dto

import "encoding/json"

// ============ 主题响应 ============

// ThemeResponse 主题响应
type ThemeResponse struct {
	Slug        string          `json:"slug"`
	Name        string          `json:"name"`
	IsActive    bool            `json:"is_active"`
	Config      json.RawMessage `json:"config"`
	Schema      json.RawMessage `json:"schema"`
	Menus       json.RawMessage `json:"menus"`
	Version     string          `json:"version"`
	Author      string          `json:"author"`
	Description string          `json:"description"`
	Cover       string          `json:"cover"`
	License     string          `json:"license"`
	Repo        string          `json:"repo"`
	CreatedAt   string          `json:"created_at"`
	UpdatedAt   string          `json:"updated_at"`
}

// ThemeListItem 主题列表项
type ThemeListItem struct {
	Slug          string `json:"slug"`
	Name          string `json:"name"`
	IsActive      bool   `json:"is_active"`
	Version       string `json:"version"`
	LatestVersion string `json:"latest_version,omitempty"`
	Author        string `json:"author"`
	Description   string `json:"description"`
	Cover         string `json:"cover"`
	License       string `json:"license"`
	Repo          string `json:"repo"`
	CreatedAt     string `json:"created_at"`
	UpdatedAt     string `json:"updated_at"`
}

// ThemeActivateResponse 主题激活响应
type ThemeActivateResponse struct {
	Slug    string `json:"slug"`
	Name    string `json:"name"`
	Message string `json:"message"`
}

// ThemeConfigUpdateRequest 更新主题配置请求
type ThemeConfigUpdateRequest struct {
	Config json.RawMessage `json:"config" binding:"required"`
}

// ThemeSchemaResponse 主题 Schema 响应（公开）
type ThemeSchemaResponse struct {
	Slug   string          `json:"slug"`
	Name   string          `json:"name"`
	Schema json.RawMessage `json:"schema"`
	Config json.RawMessage `json:"config"`
	Menus  json.RawMessage `json:"menus"`
}

// ThemeInstallResponse 主题安装响应
type ThemeInstallResponse struct {
	Slug     string `json:"slug"`
	Name     string `json:"name"`
	Version  string `json:"version"`
	IsActive bool   `json:"is_active"`
	Message  string `json:"message"`
}

// ============ 菜单数据 ============

// MenuDataItem 菜单项
type MenuDataItem struct {
	ID        uint           `json:"id"`
	Title     string         `json:"title"`
	URL       string         `json:"url"`
	Icon      string         `json:"icon"`
	Sort      int            `json:"sort"`
	IsEnabled bool           `json:"is_enabled"`
	Children  []MenuDataItem `json:"children"`
}

// MenuUpdateRequest 更新菜单请求
type MenuUpdateRequest struct {
	Type  string         `json:"type" binding:"required,max=50"`
	Items []MenuDataItem `json:"items" binding:"required"`
}

// ============ 任务状态 ============

// TaskStatusResponse 任务状态响应（主题安装/升级/激活/重建）
type TaskStatusResponse struct {
	Target  string `json:"target"`
	Status  string `json:"status"`
	Message string `json:"message"`
}

File: admin/src/types/theme.ts
// ============ 菜单类型 ============

// 菜单项（JSON 存储结构，嵌套 children）
export interface MenuDataItem {
  id: number;
  title: string;
  url: string;
  icon: string;
  sort: number;
  is_enabled: boolean;
  children: MenuDataItem[];
}

// 更新菜单请求（整体替换某个 type 的菜单列表）
export interface MenuUpdateRequest {
  type: string;
  items: MenuDataItem[];
}

// 主题菜单槽位声明（从 schema 中解析）
export interface MenuSlot {
  key: string;
  label: string;
  maxDepth: number;
}

// 主题列表项
export interface ThemeListItem {
  slug: string;
  name: string;
  is_active: boolean;
  version: string;
  latest_version?: string;
  author: string;
  description: string;
  cover: string;
  license: string;
  repo: string;
  created_at: string;
  updated_at: string;
}

/** 主题 Schema 中的单个字段定义 */
export interface SchemaField {
  type: 'string' | 'number' | 'integer' | 'boolean' | 'array' | 'object';
  title?: string;
  description?: string;
  default?: unknown;
  enum?: string[] | Array<{ label: string; value: string }>;
  format?: string;
  placeholder?: string;
  width?: number;
  height?: number;
  min?: number;
  max?: number;
  'x-item-fields'?: Array<string | Record<string, unknown>>;
}

/** 主题 Schema：允许任意嵌套分组，叶子节点为 SchemaField */
export interface ThemeSchema {
  [key: string]: SchemaField | ThemeSchema;
}

// 主题详情
export interface Theme {
  slug: string;
  name: string;
  is_active: boolean;
  author: string;
  description: string;
  cover: string;
  license: string;
  repo: string;
  config: Record<string, unknown>;
  schema: ThemeSchema;
  menus: Record<string, MenuDataItem[]>;
  version: string;
  created_at: string;
  updated_at: string;
}

// 主题安装响应
export interface ThemeInstallResponse {
  slug: string;
  name: string;
  version: string;
  is_active: boolean;
  message: string;
}

// 主题激活响应
export interface ThemeActivateResponse {
  slug: string;
  name: string;
  message: string;
}

// 主题 Schema 响应
export interface ThemeSchemaResponse {
  slug: string;
  name: string;
  schema: ThemeSchema;
}

// 主题配置更新请求
export interface ThemeConfigUpdateRequest {
  config: Record<string, unknown>;
}
